documentation/src/test/java/example/TestInfoDemo.java

```diff
@@ -24,8 +24,7 @@
 class TestInfoDemo {
 
 	TestInfoDemo(TestInfo testInfo) {
-		String displayName = testInfo.getDisplayName();
-		assertTrue(displayName.equals("TEST 1") || displayName.equals("test2()"));
+		assertEquals("TestInfo Demo", testInfo.getDisplayName());
```

Comment:
My last changes (6cda95fcfab7d6b0ca75e8a8f7dc77c246899dbe) also reverted the behavior of `TestInfo`, as the related extension doesn't have our new annotation. I therefore also reverted the `TestInfo`-related changes in `ParameterResolverTests` and `ParameterizedTestIntegrationTests`.

`TestInfo` and `TestReporter` could also benefit from the test-specific information, but these specific cases are probably not very important right now.

Reply:
> `TestInfo` and `TestReporter` could also benefit from the test-specific information, but these specific cases are probably not very important right now.

I've made a note to look at the core extensions after this PR is merged.